docs/proposals/mapper-design.md

```diff
@@ -0,0 +1,117 @@
+---
+title: Mapper Design
+status: Pending
+authors:
+    - "@sids-b"
+    - "@m1093782566"
+approvers:
+  - "@qizha"
+  - "@rohitsardesai83"
+creation-date: 2019-02-23
+last-updated: 2019-02-24
+---
+
+# Mapper Design
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Motivation](#motivation)
+  * [Goals](#goals)
+  * [Non\-goals](#non-goals)
+  * [User cases](#user-cases)
+* [Proposal](#proposal)
+
+## Introduction
+Mapper is an application that that is used to connect and and control devices. Following are the responsibilities of mapper:
+1) Scan and connect to the device.
+2) Report the actual state of twin-attributes of device.
+3) Map the expected state of device-twin to actual state of device-twin.
+4) Collect telemetry data from device.
+5) Convert readings from device to format accepted by KubeEdge(beehive message).
```

Comment:
not sure if it has to be a beehive message. needs more discussion on the telemetry data flow design overall. how event readings would be consumed by apps interacting with the platform , whether they interact directly with a database ( say influx db ? ) where our platform saves all the data ?

Reply:
My bad, It should have been message understood by platform. For each operation like memUpdate, stateUpdate, there are different structures defied by kubeedge.

Reply:
For current implementation , there is no special handling of telemetry data. It is considered in twin I guess.

Reply:
Done, Thanks